nomad/fsm.go

```diff
@@ -1629,6 +1642,21 @@ func (n *nomadFSM) reconcileQueuedAllocations(index uint64) error {
 	return nil
 }
 
+func (n *nomadFSM) applyUpsertScalingEvent(buf []byte, index uint64) interface{} {
+	defer metrics.MeasureSince([]string{"nomad", "fsm", "upsert_scaling_event"}, time.Now())
+	var req structs.ScalingEventRequest
+	if err := structs.Decode(buf, &req); err != nil {
+		panic(fmt.Errorf("failed to decode request: %v", err))
```

Comment:
do we want to panic here? could we just fail the scaling event request and log the error?

Reply:
this is the standard behavior for applying raft log entries to the state store. see, e.g.,: 
https://github.com/hashicorp/nomad/blob/88ff339f4d5bf461229e4bff018ef108fbb12d0a/nomad/fsm.go#L306-L311

it likely means that the raft log entry doesn't contain what the leading byte says it should contain, which likely means a version mismatch or corruption. the reason for the panic, i assume, is that state is undefined if we don't know how to apply a raft entry.